pkg/apis/core/v1alpha1/types_propagationpolicy.go

```diff
@@ -187,3 +193,49 @@ type ReplicaRescheduling struct {
 	// +kubebuilder:default:=true
 	AvoidDisruption bool `json:"avoidDisruption"`
 }
+
+// ReschedulePolicy determines the options regarding rescheduling.
+type ReschedulePolicy struct {
+	// DisableRescheduling determines if a federated object can be rescheduled.
+	// +optional
+	DisableRescheduling bool `json:"disableRescheduling"`
+	// When the related objects should be subject to reschedule.
+	// +optional
+	Trigger RescheduleTrigger `json:"rescheduleWhen"`
+	// Configures behaviors related to replica rescheduling.
+	// +optional
+	// Default set via a post-generation patch.
+	// See patch file for details.
+	ReplicaRescheduling *ReplicaRescheduling `json:"replicaRescheduling,omitempty"`
+}
+
+// RescheduleTrigger is a criteria for determining when the templates are subject to reschedule.
+type RescheduleTrigger struct {
+	// If set to true, the scheduler will trigger a rescheduling when the semantics of the policy changed. For example,
+	// modifying placement, schedulingMode, maxClusters, clusterSelector and other configurations related to scheduling
+	// and clustering will immediately trigger a rescheduling. Modifying the label, annotation, autoMigration
+	// configuration, or reschedulePolicy itself will not trigger the rescheduling.
```

Comment:
Modifying the `reschedulePolicy` itself doesn't trigger rescheduling - is this expected?